lib/modules.nix

```
@@ -228,6 +229,28 @@ let
               within a configuration, but can be used in module imports.
             '';
           };
+
+          _module.freeformConfig = mkOption {
+            # Do not show this in the documentation
+            internal = true;
+            visible = false;
+            # Cant add because it has a default
+            readOnly = true;
+            # TODO: Implement the type
+            # type =
+            #   if options._module.freeformType.value != null
+            #   then options._module.freeformType.value
+            #   # The value is the empty set. But we dont have a type yet for that.
+            #   else (types.attrsOf types.bool);
+            # default = type.emptyValue.value or { };
+            description = ''
+              Read-only version of the freeform definitions. This includes all definitions that don't have matching options if the freeformType is set.
+              Otherwise this is an empty attribute set.
+            '';
+            # Discards the value but keeps the definition
+            # Which is then available via '.options._module.freeform.definitions'
```

The type-based option value is never computed, so it can't *technically* be discarded.
```suggestion
            # It's currently(?) not possible to compute the option value normally, using the type.
            # Instead, we set it directly here, without evaluating the merge result from the type.
```